### PR TITLE
Add `tzarc/ghoul`.

### PR DESCRIPTION
**Pull request by tzarc**
## Description

As per the title.

PID/VID combo is valid: https://pid.codes/1209/4920/

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tzarc** [comment]
CI build error is because `qmk list-keyboards` incorrectly detects the base folder as a keyboard due to the presence of `rules.mk`.

**Comment by tzarc** [comment]
Needs a `readme.md`. Incoming.

File: keyboards/tzarc/ghoul/ghoul.h [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2018-2022 Nick Brassel (@tzarc)
```
Comment (zvecr) [inline keyboards/tzarc/ghoul/ghoul.h:1]: This file can be removed as DD layout macro

File: keyboards/tzarc/ghoul/info.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  "matrix_pins": {
+    "rows": ["NO_PIN","NO_PIN","NO_PIN","NO_PIN","NO_PIN","NO_PIN"],
+    "cols": ["NO_PIN","NO_PIN","NO_PIN","NO_PIN","NO_PIN","NO_PIN","NO_PIN","NO_PIN"]
+  },
+  "usb": {
+    "vid": "0x1209",
+    "pid": "0x4920"
+  },
+  "layouts": {
+    "LAYOUT_all": {
```
Comment (zvecr) [inline keyboards/tzarc/ghoul/info.json:27]:
```suggestion
    "LAYOUT": {
```

File: keyboards/tzarc/ghoul/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+#define HOME_SCLN RGUI_T(KC_SCLN)
+
+#define SFT_ESC LSFT_T(KC_ESC)
+
+extern void ui_init(void);
+extern void ui_task(void);
+
+// clang-format off
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_QWERTY] = LAYOUT_all(
```
Comment (zvecr) [inline keyboards/tzarc/ghoul/keymaps/default/keymap.c:29]:
```suggestion
    [_QWERTY] = LAYOUT(
```

File: keyboards/tzarc/ghoul/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,            RGB_MOD,            KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,
+        HOME_A,  HOME_S,  HOME_D,  HOME_F,  KC_G,                                KC_H,    HOME_J,  HOME_K,  HOME_L,  HOME_SCLN,
+        KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,                                KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,
+        SFT_ESC, KC_LCTL, KC_LGUI, KC_SPC,  LOWER,                               RAISE,   KC_SPC,  KC_LALT, KC_BSPC, KC_SFTENT
+    ),
+    [_LOWER] = LAYOUT_all(
```
Comment (zvecr) [inline keyboards/tzarc/ghoul/keymaps/default/keymap.c:35]:
```suggestion
    [_LOWER] = LAYOUT(
```

File: keyboards/tzarc/ghoul/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,         RGB_MOD,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                             KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                             KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                             KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [_RAISE] = LAYOUT_all(
```
Comment (zvecr) [inline keyboards/tzarc/ghoul/keymaps/default/keymap.c:41]:
```suggestion
    [_RAISE] = LAYOUT(
```

File: keyboards/tzarc/ghoul/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,         RGB_MOD,            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                             KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                             KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                             KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [_ADJUST] = LAYOUT_all(
```
Comment (zvecr) [inline keyboards/tzarc/ghoul/keymaps/default/keymap.c:47]:
```suggestion
    [_ADJUST] = LAYOUT(
```

File: keyboards/tzarc/ghoul/readme.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# Ghoul
+
+![Ghoul](https://i.imgur.com/T8wp7cE.jpg)
+
+*A short description of the keyboard/project*
```
Comment (zvecr) [inline keyboards/tzarc/ghoul/readme.md:5]:
```suggestion
Planck-like development board for QMK, using SparkFun MicroMod for hot-swap MCU capability.
```

File: keyboards/tzarc/ghoul/readme.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+* Keyboard Maintainer: [tzarc](https://github.com/tzarc)
+* Hardware Supported: Ghoul rev1 PCB + MicroMod STM32, MicroMod RP2040
+Hardware Availability: [Open-source](https://github.com/tzarc/ghoul)
```
Comment (zvecr) [inline keyboards/tzarc/ghoul/readme.md:9]:
```suggestion
* Hardware Availability: [Open-source](https://github.com/tzarc/ghoul)
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/tzarc/ghoul/readme.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
+# Ghoul
+
+![Ghoul](https://i.imgur.com/T8wp7cE.jpg)
```
Comment (drashna) [inline keyboards/tzarc/ghoul/readme.md:3]:
```suggestion
![Ghoul](https://i.imgur.com/T8wp7cEh.jpg)
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**